File: server/internal/model/conversation.go
// server/internal/model/conversation.go
package model

import "time"

// Conversation 会话表，管理一次连贯的对话生命周期
type Conversation struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	UserID       uint      `gorm:"index" json:"user_id"`
	PortfolioID  uint      `gorm:"index" json:"portfolio_id"`
	Title        string    `gorm:"size:200" json:"title"`
	ModelName    string    `gorm:"size:50" json:"model_name"`
	Summary      string    `gorm:"type:text" json:"summary"`
	MessageCount int       `json:"message_count"`
	Status       string    `gorm:"size:20;default:active" json:"status"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

// 会话状态常量
const (
	ConversationStatusActive   = "active"
	ConversationStatusArchived = "archived"
)

// Message 消息表，记录每一轮对话
type Message struct {
	ID             uint      `gorm:"primaryKey" json:"id"`
	ConversationID uint      `gorm:"index" json:"conversation_id"`
	Role           string    `gorm:"size:20" json:"role"` // user, assistant, system
	Content        string    `gorm:"type:text" json:"content"`
	TokenCount     int       `json:"token_count"`
	TaskID         *uint     `json:"task_id"` // 关联的 AITask（可选）
	CreatedAt      time.Time `json:"created_at"`
}

File: web/src/api/conversation.ts
// web/src/api/conversation.ts
import request from './request'

export interface Conversation {
  id: number
  user_id: number
  portfolio_id: number
  title: string
  model_name: string
  summary: string
  message_count: number
  status: string
  created_at: string
  updated_at: string
}

export interface Message {
  id: number
  conversation_id: number
  role: 'user' | 'assistant' | 'system'
  content: string
  token_count: number
  task_id: number | null
  created_at: string
}

export interface CreateConversationRequest {
  portfolio_id: number
  model_name?: string
  title?: string
}

export interface SendMessageRequest {
  content: string
}

export const conversationApi = {
  create: (data: CreateConversationRequest) =>
    request.post('/conversations', data),

  list: (params?: { portfolio_id?: number; page?: number; page_size?: number }) =>
    request.get('/conversations', { params }),

  get: (id: number) =>
    request.get(`/conversations/${id}`),

  sendMessage: (id: number, data: SendMessageRequest) =>
    request.post(`/conversations/${id}/messages`, data),

  archive: (id: number) =>
    request.delete(`/conversations/${id}`),
}
